cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/documentstore/DatabaseImpl.java

```diff
@@ -1016,6 +1053,14 @@ public DocumentRevision delete(final DocumentRevision rev) throws
     public List<DocumentRevision> delete(final String id)
             throws DocumentStoreException {
         Misc.checkNotNull(id, "ID");
+        // check for local prefix:
+        // there are two reasons not to delete local documents here:
+        // 1) "delete all leaf nodes" has no meaning for local documents
+        // 2) DeleteAllRevisionsCallable can throw DocumentNotFoundException which would mean a
```

Comment:
I'm not sure I quite follow this.
`db.delete("_local/mydoc")` seems a lot more intuitive than `db.delete(new DocumentRevisions("_local/mydoc",null))` so it might be nice to have this even though the return of an empty list is a little weird for a local doc.
Are you saying that we don't expose `DocumentNotFoundException` on the signature of this method, although we do for `delete(DocumentRev)`?
It seems to me that `DeleteAllRevisionsCallable` throwing a `DocumentNotFoundException` results in a `DocumentStoreException`, but you're reluctant to have this same behaviour for deleting the local doc?

Reply:
OK, I've changed this in 5fac0be so that this flavour also deletes local documents, and changed the method signature as discussed